python/Dockerfile

```diff
@@ -14,6 +14,8 @@ ARG PY_3_8
 ARG PYENV_VERSION
 USER root
 
+# Install *only* the apt packages required for this builder image to build Python.
+# C-libs needed by users to build their Python packages should be installed down below in the final docker image.
```

Comment:
This adds a c-lib here:
* https://github.com/dependabot/dependabot-core/pull/7875

Did it as a separate PR to prove CI is okay with this first step individually, separate from the additional install.